src/v/storage/segment_appender.h

```diff
@@ -77,6 +77,20 @@ class segment_appender {
 
     void set_callbacks(callbacks* callbacks) { _callbacks = callbacks; }
 
+    /** Validator for fallocation step configuration setting */
+    static std::optional<ss::sstring>
+    validate_fallocation_step(const size_t& value) {
```

Comment:
nit: there is not any benefit to taking a `size_t` by `const&`. I wonder if `config::` is making this requirement?